### PR TITLE
docs(components): [switch] use new display tag

### PR DESCRIPTION
**Pull request by wzc520pyfm**
Related: #10754 

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/switch/src/switch.vue [hunk 1]
```diff
@@ -120,6 +112,39 @@ useDeprecated(
...
+  {
+    from: '"border-color"',
+    replacement: 'CSS var `--el-switch-border-color`',
+    scope: COMPONENT_NAME,
+    version: '2.3.0',
+    ref: 'https://element-plus.org/en-US/component/switch.html#attributes',
+    type: 'Attribute',
+  },
+  computed(() => !!vm.vnode.props?.borderColor)
+)
```
Comment (btea) [inline packages/components/switch/src/switch.vue:147]: There's some repetitive code here, perhaps it can be extracted. 🤔

Comment (wzc520pyfm) [reply packages/components/switch/src/switch.vue:147]: What do you think of the following?
```js
type DeprecationTypes = Parameters<typeof useDeprecated>
const useBatchDeprecated = (
  params: Array<
    Pick<DeprecationTypes[0], 'from' | 'replacement'> & {
      condition: DeprecationTypes[1]
    }
  >
) => {
  params.forEach((param) => {
    useDeprecated(
      {
        from: param.from,
        replacement: param.replacement,
        scope: COMPONENT_NAME,
        version: '2.3.0',
        ref: 'https://element-plus.org/en-US/component/switch.html#attributes',
        type: 'Attribute',
      },
      param.condition
    )
  })
}
useBatchDeprecated([
  {
    from: '"value"',
    replacement: '"model-value" or "v-model"',
    condition: computed(() => !!vm.vnode.props?.value),
  },
  {
    from: '"active-color"',
    replacement: 'CSS var `--el-switch-on-color`',
    condition: computed(() => !!vm.vnode.props?.activeColor),
  },
  {
    from: '"inactive-color"',
    replacement: 'CSS var `--el-switch-off-color`',
    condition: computed(() => !!vm.vnode.props?.inactiveColor),
  },
  {
    from: '"border-color"',
    replacement: 'CSS var `--el-switch-border-color`',
    condition: computed(() => !!vm.vnode.props?.borderColor),
  },
])
```

Comment (btea) [reply packages/components/switch/src/switch.vue:147]: Maybe this is more concise: 
```ts
const useBatchDeprecated = (list: string[][]) => {
  list.forEach((param) => {
    useDeprecated(
      {
        from: param[0],
        replacement: param[1],
        scope: COMPONENT_NAME,
        version: '2.3.0',
        ref: 'https://element-plus.org/en-US/component/switch.html#attributes',
        type: 'Attribute',
      },
      computed(() => !!vm.vnode.props?[param[2]])
    )
  })
}
useBatchDeprecated([
 ['value', '"model-value" or "v-model"', 'value'],
 ...
])
```

Comment (wzc520pyfm) [reply packages/components/switch/src/switch.vue]: updated

**Review by btea — APPROVED** [review]

File: docs/en-US/component/switch.md [hunk 2]
```diff
@@ -79,37 +79,42 @@ switch/prevent-switching
...
+| before-change                | before-change hook before the switch state changes. If `false` is returned or a `Promise` is returned and then is rejected, will stop switching | ^[boolean] / ^[Function]`() => Promise<boolean>` | —       |
+| id                           | id for input                                                                                                                                    | ^[string]                                        | —       |
+| tabindex                     | tabindex for input                                                                                                                              | ^[string] / ^[number]                            | —       |
+| value ^(deprecated)          | binding value ( deprecated, use `model-value / v-model` instead )                                                                               | ^[boolean] / ^[string] / ^[number]               | false   |
```
Comment (tolking) [inline docs/en-US/component/switch.md:108]: Why add deprecated attributes to the document?

Comment (wzc520pyfm) [reply docs/en-US/component/switch.md:108]: I found that the 'value' property is still included in the switch property declaration, so I added it. Let me remove it?

Comment (tolking) [reply docs/en-US/component/switch.md:108]: I don't feel the need to add it to the docs.

Comment (wzc520pyfm) [reply docs/en-US/component/switch.md]: I agree, I have deleted it from the docs.

**Review by tolking — APPROVED** [review]

**Pull request merged**